src/driver-marsgaming/marsgaming-leds.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+#include "stdint.h"
+#include "libratbag.h"
+
+enum marsgaming_led_brightness_intensity {
+	MARSGAMING_LED_BRIGHTNESS_OFF = 0x00,
+	MARSGAMING_LED_BRIGHTNESS_LOW = 0x01,
+	MARSGAMING_LED_BRIGHTNESS_MEDIUM = 0x02,
+	MARSGAMING_LED_BRIGHTNESS_HIGH = 0x03
+};
+
+#define MARSGAMING_LED_BREATHING_OFF 0
+
+struct marsgaming_led_color {
+	uint8_t red; // 0xff - red
+	uint8_t green; // 0xff - green
+	uint8_t blue; // 0xff - blue
+};
+
+struct marsgaming_led {
+	struct marsgaming_led_color color;
+	uint8_t brightness; // marsgaming_led_brightness_intensity
+	uint8_t breathing_speed; // marsgaming_led_breathing_speed
```

Comment:
I think these should have these enums as the type, although I'm not 100%.
(would be nice if someone more experienced in C could confirm this).

Reply:
Also, there is no `enum marsgaming_led_breathing_speed`.

Reply:
I tried using enums, but I didn't find any way of specifying the backing type for the enum, so using it in usb reports directly might make this code platform-dependent.

Reply:
You need `packed` attribute and an assertion to do a size check.
Here is an example of an enum from the sinowealth driver:
```c
enum sinowealth_report_id {
	SINOWEALTH_REPORT_ID_CONFIG = 0x4,
	SINOWEALTH_REPORT_ID_CMD = 0x5,
	SINOWEALTH_REPORT_ID_CONFIG_LONG = 0x6,
} __attribute__((packed));
_Static_assert(sizeof(enum sinowealth_report_id) == sizeof(uint8_t), "Invalid size");
```

Reply:
> By the way, there are so many commits almost to the point of spam, like almost 10 instances `Marsgaming Return type on new line`, does this imply you are expecting the commits squashed? (They don't appear to be in form of incremental changes anyway)

Yes, I was expecting the commits to be squashed.